distributed/worker_state_machine.py

```diff
@@ -1963,6 +1964,8 @@ def _transition_from_resumed(
 
             if ts._previous in ("executing", "long-running"):
                 self._release_resources(ts)
+                self.executing.discard(ts)
+                self.long_running.discard(ts)
```

Comment:
This is actually superfluous as we're invoking purge_state afterwards. But I think it's a good idea to have it for cleanliness' sake.